tests/Smoke/InstallViaComposerTest.php

```diff
@@ -164,8 +157,92 @@ public function testInstallationViaArtifactIsPossible(): void
         self::assertCommandsWork($stepsToPrepareArtifact, $tmpArtifactPath);
         self::assertCommandsWork($this->stepsToVerifyInstallation, $tmpPath);
 
-        $fs->remove($tmpPath);
-        $fs->remove($tmpArtifactPath);
+        $this->fs->remove($tmpPath);
+        $this->fs->remove($tmpArtifactPath);
+    }
+
+    /**
+     * @param array<string> $additionalCommands
+     *
+     * @dataProvider provideDoctrineAnnotationRulesetThrowsAnExceptionWhenSupportedDoctrinePackagesAreNotInstalledCases
+     */
+    public function testDoctrineAnnotationRulesetThrowsAnExceptionWhenSupportedDoctrinePackagesAreNotInstalled(array $additionalCommands): void
+    {
+        $tmpPath = $this->createFakeComposerProject($this->currentCodeAsComposerDependency);
+
+        self::assertCommandsWork(
+            [
+                'composer install -q',
+                'composer dump-autoload --optimize',
```

Comment:
What is the reason for this to be separate instead of adding `--optimize-autoloader` to `composer install`?

Reply:
We need to ask Mr. Copy Paste 😅.